File: Array_Operations/removeElement.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int n=nums.length,temp=0;
        for(int i=0;i<n;i++)
        {
            if(nums[i]!=val)
            {
                nums[temp++]=nums[i];
            }

        }
        return temp;
    }
}
public class main()
{
    public static void main(String args[])
    {
        Solution solution=new Solution();
        int[] num = new int[]{0,1,2,2,3,0,4,2};
        solution.removeElement(num,2);
    }
}

File: Array_Operations/removeElement.py
class Solution(object):
    def removeElement(self, nums, val):
        '''counter = 0
        for i in range(len(nums)):
            if nums[i] == val:
                temp = nums[i]
                for j in range(i, len(nums)-counter-1):
                    nums[j] = nums[j+1]
                    counter = counter + 1
                nums[j] = temp
                
        return counter-1'''
        i=0
        for j in range(len(nums)):
            if nums[j]!=val:
                nums[i]=nums[j]
                i = i+1
                
        return i

S = Solution()
nums = [0,1,2,2,3,0,4,2]
count = S.removeElement(nums,2)
for i in range(count):
    print(nums[i])